src/spatialNavigation.js

```diff
@@ -51,6 +51,9 @@ export const getChildClosestToOrigin = (children) => {
   return first(childrenClosestToOrigin);
 };
 
+// Test for presence of DOM to avoid errors in SSR
+const hasDOM = typeof window !== 'undefined';
```

Comment:
I'm thinking now that this check is not sufficient, also in the previous PR.
For example in the native environments when using react native, there is a window object, but it doesn't have methods like "addEventListener", or "removeEventListener", or "document", so it wouldn't work like this. All the checks should be the same as they were before with the only difference of checking "typeof undefined" instead of just "window".

Reply:
Agreed.
At present I am only experiencing this in server side situations so just dealing with this there.
Would a check for `document` suffice?

Reply:
I think the safest would be to replace only the `window` check with the `typeof window !== 'undefined'` but still leave the second part of the checks as it was.
For example:
`window && window.document` => `typeof window !== 'undefined' && window.document`
For example in react native they are adding mocks for global variables, so even the "document" could be defined, but there would be no "addEventListener" for example.
So yes, checking in relevance to the context is the best. Could you also add the additional `&& window.document` to the check in the previous PR please?
Just to be sure it's as backward compatible as possible 👍

Reply:
Will do.